### ISSUE TITLE
Troubles with installing my private package from private S3 index built with S3PyPi

### ISSUE THREAD

**Issue by Redhouane** [novemberfiveco/s3pypi#58]
Hi all,
First of all I would lie to thank you for this wonderful tool. The built of a private S3 index was so easy using S3PyPi (following [your tutorial](https://my.su/ug9l2)).

Actually I'm trying to install a homemade package using `pipenv` from my private S3 pypi index index.

I'm using:

- python `3.7.3`
- pip `19.2.1`
- pipenv `2018.11.26`

The install command line:
`pipenv install my-homemade-package --extra-index-url https://my-private-index`

I'm facing the following issue below and do not understand what I'm doing wrong (since the final issue did mension an unreachable).

```
Installing my-homemade-package…
Adding my-homemade-package to Pipfile's [packages]…
✔ Installation Succeeded 
Pipfile.lock (67b9a8) out of date, updating to (8e084c)…
Locking [dev-packages] dependencies…
✔ Success! 
Locking [packages] dependencies…
✘ Locking Failed! 
[pipenv.exceptions.ResolutionFailure]:   File "/home/redhouane/.local/lib/python3.7/site-packages/pipenv/resolver.py", line 69, in resolve
[pipenv.exceptions.ResolutionFailure]:       req_dir=requirements_dir
[pipenv.exceptions.ResolutionFailure]:   File "/home/redhouane/.local/lib/python3.7/site-packages/pipenv/utils.py", line 726, in resolve_deps
[pipenv.exceptions.ResolutionFailure]:       req_dir=req_dir,
[pipenv.exceptions.ResolutionFailure]:   File "/home/redhouane/.local/lib/python3.7/site-packages/pipenv/utils.py", line 480, in actually_resolve_deps
[pipenv.exceptions.ResolutionFailure]:       resolved_tree = resolver.resolve()
[pipenv.exceptions.ResolutionFailure]:   File "/home/redhouane/.local/lib/python3.7/site-packages/pipenv/utils.py", line 395, in resolve
[pipenv.exceptions.ResolutionFailure]:       raise ResolutionFailure(message=str(e))
[pipenv.exceptions.ResolutionFailure]:       pipenv.exceptions.ResolutionFailure: ERROR: ERROR: Could not find a version that matches my-homemade-package
[pipenv.exceptions.ResolutionFailure]:       No versions found
[pipenv.exceptions.ResolutionFailure]: Warning: Your dependencies could not be resolved. You likely have a mismatch in your sub-dependencies.
  First try clearing your dependency cache with $ pipenv lock --clear, then try the original command again.
 Alternatively, you can use $ pipenv install --skip-lock to bypass this mechanism, then run $ pipenv graph to inspect the situation.
  Hint: try $ pipenv lock --pre if it is a pre-release dependency.
ERROR: ERROR: Could not find a version that matches my-homemade-package
No versions found
Were https://pypi.org/simple or https://my-private-index.my-domaine.com/my-homemade-package/ reachable?
[pipenv.exceptions.ResolutionFailure]:       req_dir=requirements_dir
[pipenv.exceptions.ResolutionFailure]:   File "/home/redhouane/.local/lib/python3.7/site-packages/pipenv/utils.py", line 726, in resolve_deps
[pipenv.exceptions.ResolutionFailure]:       req_dir=req_dir,
[pipenv.exceptions.ResolutionFailure]:   File "/home/redhouane/.local/lib/python3.7/site-packages/pipenv/utils.py", line 480, in actually_resolve_deps
[pipenv.exceptions.ResolutionFailure]:       resolved_tree = resolver.resolve()
[pipenv.exceptions.ResolutionFailure]:   File "/home/redhouane/.local/lib/python3.7/site-packages/pipenv/utils.py", line 395, in resolve
[pipenv.exceptions.ResolutionFailure]:       raise ResolutionFailure(message=str(e))
[pipenv.exceptions.ResolutionFailure]:       pipenv.exceptions.ResolutionFailure: ERROR: ERROR: Could not find a version that matches my-homemade-package
[pipenv.exceptions.ResolutionFailure]:       No versions found
[pipenv.exceptions.ResolutionFailure]: Warning: Your dependencies could not be resolved. You likely have a mismatch in your sub-dependencies.
  First try clearing your dependency cache with $ pipenv lock --clear, then try the original command again.
 Alternatively, you can use $ pipenv install --skip-lock to bypass this mechanism, then run $ pipenv graph to inspect the situation.
  Hint: try $ pipenv lock --pre if it is a pre-release dependency.
ERROR: ERROR: Could not find a version that matches my-homemade-package
No versions found
Were https://pypi.org/simple or https://my-private-index.my-domaine.com/my-homemade-package/ reachable?
```
I did try the approaches recommanded in the stack trace : 

First:
1 - Try clearing your dependency cache with `pipenv lock --clear`
2 - Try the original command again (the issue persist)

Second:
1 - Use `pipenv install --skip-lock` to bypass this mechanism
2 - run `pipenv graph` to inspect the situation (No dependency related to my private package is listed)

Third:
Try `pipenv lock --pre` if it is a pre-release dependency 

Fourth:
I tryied to install my homemade package from my private package using `pip` with the following commande line:
`pip install my-homemade-package --extra-index-url https://my-private-index.my-domaine.com/`
But still does not work with the following error message:

```
WARNING: Retrying (Retry(total=4, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.VerifiedHTTPSConnection object at 0x7f9b67abb278>: Failed to establish a new connection: [Errno -2] Name or service not known')': /my-homemade-package/
WARNING: Retrying (Retry(total=3, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.VerifiedHTTPSConnection object at 0x7f9b67abb550>: Failed to establish a new connection: [Errno -2] Name or service not known')': /my-homemade-package/
WARNING: Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.VerifiedHTTPSConnection object at 0x7f9b67d0e5c0>: Failed to establish a new connection: [Errno -2] Name or service not known')': /my-homemade-package/
WARNING: Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.VerifiedHTTPSConnection object at 0x7f9b67a9fef0>: Failed to establish a new connection: [Errno -2] Name or service not known')': /my-homemade-package/
WARNING: Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.VerifiedHTTPSConnection object at 0x7f9b67a9ff98>: Failed to establish a new connection: [Errno -2] Name or service not known')': /my-homemade-package/
ERROR: Could not find a version that satisfies the requirement my-homemade-package (from versions: none)
ERROR: No matching distribution found for my-homemade-package
```

P.S: I did try to add my private index url into enabled source within my `Pipefile` with the following lines:

```
[[source]]
url = "https://my-private-index.my-domaine.com/"
verify_ssl = true
name = "my-private-index"
```
but, one more time, the issue still persist.

Did you ever fecad a similar issue ?
Thx in advance for your help,

Red1

**Comment by sidp8** [comment]
I am having the same issues,
Hoping someone found the solution

*error - ERROR: Could not find a version that satisfies the requirement test (from versions: none)
ERROR: No matching distribution found for test*

Whist this happens most of the times i try , sometimes , it does install but the modules , attributes do not load.

Thanks in advance for the help

**Comment by sad-boy-uoa** [comment]
Hey guys, have a look at #35

**Comment by mdwint** [comment]
This is likely fixed in s3pypi 1.0.0.
